src/core/algorithms/dd/dd_verifier/dd_verifier.cpp

```diff
@@ -0,0 +1,211 @@
+}
+
+void DDVerifier::MakeExecuteOptsAvailable() {
+    using namespace config::names;
+    MakeOptionsAvailable({kDDString});
+}
+
+void DDVerifier::CheckCorrectnessDd() const {
+    for (auto const &constraint : dd_.left) {
+        assert(constraint.constraint.upper_bound >= constraint.constraint.lower_bound);
+        assert(constraint.constraint.lower_bound >= 0);
+        assert(constraint.constraint.upper_bound >= 0);
+    }
+    for (auto const &constraint : dd_.right) {
+        assert(constraint.constraint.upper_bound >= constraint.constraint.lower_bound);
+        assert(constraint.constraint.lower_bound >= 0);
+        assert(constraint.constraint.upper_bound >= 0);
+    }
+}
+
+unsigned long long DDVerifier::ExecuteInternal() {
+    std::vector<model::ColumnIndex> columns;
+    for (auto const &constraint : dd_.left) {
+        model::ColumnIndex column_index =
+                typed_relation_->GetSchema()->GetColumn(constraint.column_name)->GetIndex();
+        columns.push_back(column_index);
+    }
+    for (auto const &constraint : dd_.right) {
+        model::ColumnIndex column_index =
+                typed_relation_->GetSchema()->GetColumn(constraint.column_name)->GetIndex();
+        columns.push_back(column_index);
+    }
+    for (auto const col : columns) {
+        assert(IsColumnMetrizable(col));
+    }
+    CheckCorrectnessDd();
+    num_rows_ = typed_relation_->GetNumRows();
+    num_columns_ = typed_relation_->GetNumColumns();
+
+    auto start_time = std::chrono::system_clock::now();
+
+    VerifyDD();
+
+    PrintStatistics();
+
+    auto elapsed_milliseconds = std::chrono::duration_cast<std::chrono::milliseconds>(
+            std::chrono::system_clock::now() - start_time);
+
+    return elapsed_milliseconds.count();
+}
+
+void DDVerifier::CheckDFOnRhs(std::vector<std::pair<int, int>> const &lhs) {
+    std::vector<model::ColumnIndex> columns;
+    for (auto const &dd : dd_.right) {
+        model::ColumnIndex column_index =
+                typed_relation_->GetSchema()->GetColumn(dd.column_name)->GetIndex();
+        columns.push_back(column_index);
+    }
+    for (auto const &pair : lhs) {
+        auto curr_constraint = dd_.right.cbegin();
+        bool is_error = false;
+        for (auto const column_index : columns) {
+            if (double const dif = CalculateDistance(column_index, pair);
+                !curr_constraint->constraint.Contains(dif)) {
+                highlights_.emplace_back(column_index, pair);
+                is_error = true;
+            }
+            ++curr_constraint;
+        }
+        if (is_error) {
+            ++num_error_rhs_;
+        }
+    }
+}
+
+void DDVerifier::VerifyDD() {
+    std::vector<std::pair<int, int>> const lhs = GetRowsWhereLhsHolds();
+    CheckDFOnRhs(lhs);
+    if (lhs.empty()) {
+        error_ = 0;
+    } else {
+        error_ = static_cast<double>(num_error_rhs_) / static_cast<double>(lhs.size());
+    }
```

Comment:
We don't have to explicitly cast **both** values to `double`, usually casting the dividend is enough:
```cpp
    error_ = lhs.empty() ? 0 : static_cast<double>(num_error_rhs_) / lhs.size();
```